contracts/voting/Voting.sol

```diff
@@ -0,0 +1,350 @@
+    }
+
+    function _startVoting(uint256 regionId) internal {
+        // solhint-disable-next-line not-rely-on-time
+        uint256 currentTimestamp = block.timestamp;
+        if (_regions[regionId].startVotingTimestamp == 0) {
+            _regions[regionId] = Region({active: true, startVotingTimestamp: currentTimestamp});
+            emit VotingStarted(regionId, currentTimestamp);
+        }
+    }
+
+    function _seasonNumber(uint256 regionId) internal view returns(uint256) {
+        // solhint-disable-next-line not-rely-on-time
+        return ((block.timestamp - _regions[regionId].startVotingTimestamp) /
+            (VOTING_DURATION + GOVERNANCE_DURATION)) + 1;
+    }
+
+    function _isVotingPeriod(uint256 regionId) internal view returns(bool) {
+        // solhint-disable-next-line not-rely-on-time
+        return ((block.timestamp - _regions[regionId].startVotingTimestamp) %
+            (VOTING_DURATION + GOVERNANCE_DURATION)) < VOTING_DURATION;
+    }
+
+    function _isGoverningPeriod(uint256 regionId) internal view returns(bool) {
+        // solhint-disable-next-line not-rely-on-time
+        return ((block.timestamp - _regions[regionId].startVotingTimestamp) %
+            (VOTING_DURATION + GOVERNANCE_DURATION)) > VOTING_DURATION;
+    }
+
+    function _calculateVotesPrice(
+        uint256 mayorId,
+        uint256 cityId,
+        uint256 votes
+    ) internal view returns(uint256) {
+        return (votes * _cities[cityId].votePrice * _getVoteMultiplier(mayorId, cityId, msg.sender)) / 100;
+    }
+
+    function _getBuildingsDiscount(uint256 cityId, address account) internal view returns(uint256) {
+        if (_ownerToBuildings[account][cityId][Building.University] > 0) {
+            return VotingConstants.BUILDING_DISCOUNT_UNIVERSITY;
+        } else if (_ownerToBuildings[account][cityId][Building.Hospital] > 0) {
+            return VotingConstants.BUILDING_DISCOUNT_HOSPITAL;
+        } else {
+            return 0;
+        }
+    }
+
+    function _getVoteMultiplier(
+        uint256 nftId,
+        uint256 cityId,
+        address account
+    ) internal view returns(uint256) {
+        return 100 - _mayor.getVoteDiscount(nftId) - _getBuildingsDiscount(cityId, account);
+    }
+
+    function _calculateWinner(
+        uint256 season,
+        uint256 cityId
+    ) internal view returns(uint256) {
+        uint256 bank = _getBank(cityId, season);
+        if (bank == 0) revert VotingErrors.IncorrectValue();
+
+        uint256 random = uint256(keccak256(abi.encodePacked(season, cityId)));
+        uint256 winnerRate = random % bank;
+        uint256 votesCounter = 0;
+
+        Nominee[] storage nominees = _cityToNominees[cityId][season];
+        uint256 nomineesLength = nominees.length;
+        for (uint256 i = 0; i < nomineesLength; i++) {
+            votesCounter += nominees[i].votes;
+            if (winnerRate < votesCounter) {
+                return nominees[i].mayorId;
+            }
```

Comment:
as I understand the process the decision on the winner is based on their schedule. if so - that's bad since the earlier the better